### PR TITLE
Improve ScheduledSplit#hashCode

### PR DESCRIPTION
**Pull request by wenleix**
```
== NO RELEASE NOTE ==
```

### CODE REVIEW

File: presto-main/src/main/java/com/facebook/presto/execution/ScheduledSplit.java [hunk 1]
```diff
@@ -60,7 +61,7 @@ public Split getSplit()
     @Override
     public int hashCode()
     {
-        return Longs.hashCode(sequenceId);
+        return Objects.hash(planNodeId, sequenceId);
```
Comment (arhimondr) [inline presto-main/src/main/java/com/facebook/presto/execution/ScheduledSplit.java]: You need to update the `equals` then. Why are we changing that? How the `equals` / `hashcode` is used for this object? Why don't we include the `split` to the comparison?

**Comment by rschlussel** [comment]
Why were we only using the sequenceId before?

**Comment by wenleix** [comment]
It's originally for https://github.com/prestodb/presto/pull/13895 to guarantee splits will be distributed evenly across Spark tasks. But now I realized we can just compute the hashcode by the getter methods without change `ScheduledSplit#hashCode`

**Comment by arhimondr** [comment]
@wenleix 

The general `hashCode` and `equals` consistency contract is that

> If two objects are equal(according to equals() method) then the hashCode() method should return the same integer value for both the objects.

Otherwise data structures such as `HashSet` and `HashMap` may not work correctly.

Thus if `hashCode` now includes the `planNodeId` - the `equals` should also include `planNodeId`.

I see a lot of places where `ScheduledSplit` is added to a set. But since `sequenceId` is guaranteed to be unique across all splits for the same Task, I see no hard of adding `planNodeId` into the comparison.

However, i think it is safer to simply take Murmur3 / XXHash of both, sequenceId + planNodeId in the Presto Spark codebase for more uniform distribution.

**Comment by wenleix** [comment]
@arhimondr : 
> However, i think it is safer to simply take Murmur3 / XXHash of both, sequenceId + planNodeId in the Presto Spark codebase for more uniform distribution.

Correct, that's why I said we no longer need to change the hash code implementation of `ScheduledSplit` itself. I don't even think we need XXHash, just `Objects.hash` should be good enough :)

**Comment by arhimondr** [comment]
@wenleix 

> Correct, that's why I said we no longer need to change

Sorry, totally missed that. Nevermind then.

**Pull request closed without merging**